plugins/catalog-backend/src/ingestion/processors/types.ts

```diff
@@ -113,6 +126,34 @@ export type CatalogProcessorParser = (options: {
   location: LocationSpec;
 }) => AsyncIterable<CatalogProcessorResult>;
 
+/**
+ * A cache for storing data during processing.
+ *
+ * The values stored in the cache are always local to each processor, meaning
+ * no processor can see cache values from other processors.
+ *
+ * The cache instance provided to the CatalogProcessor is also scoped to the
+ * entity being processed, meaning that each processor run can't see cache
+ * values from processing runs for other entities.
+ *
+ * Values that are set during a processing run will only be visible in the directly
+ * following run. The cache will be overwritten every run, meaning existing values
+ * are removed and need to be set again for them to remain in the cache.
```

Comment:
I'm still not entirely sure about this mechanic.

Imagine starting with an empty state.

- We are called to read `url1`: no matching entry for that url was found in the cache; the remote is read and we store `{ etag, data }` in that entry in the cache
- We are called to read `url1` one more time: a matching entry for that url was found in the cache; we poll the remote using that etag and find that it still hasn't changed, we then return `data`

But at this point we are expected to do a `set` on the cache again as well? Writing `{ etag, data }` one more time? Makes one kinda wonder, why?

It would be really good if the cache also stored a last write time per entry, transparently, because we might use that to only ever try to poll the remote at most once every N seconds etc

Reply:
Storing the last write time is one of the big patterns I think this will be used for, it's essentially gonna be the way that processors can poll slower than the refresh loop. I don't think it makes sense for the base URL reading though, at least not without any additional configuration.

Regarding the mechanic, what do you propose instead? I feel that TTL is just more complicated and error prone, and making the keys persist based on some criteria is gonna be too magic. This is very explicit and still performant, but open to suggestions

Reply:
Yeah I vote TTL I think. That's what we'd do if it'd been backend by a classic cache store.

Reply:
What would that look like in more detail and what would the benefits be?

Reply:
Keeping the overall architecture, but the extra `set()` call is no longer be needed.